Sources/Rendering/OpenGL/RenderWindow/index.js

```diff
@@ -733,8 +733,16 @@ function vtkOpenGLRenderWindow(publicAPI, model) {
   };
 
   publicAPI.getHardwareMaximumLineWidth = () => {
+    // We cache the result of this function because `getParameter` is slow
+    if (model.hardwareMaximumLineWidth != null) {
+      return model.hardwareMaximumLineWidth;
+    }
+
     const gl = publicAPI.get3DContext();
     const lineWidthRange = gl.getParameter(gl.ALIASED_LINE_WIDTH_RANGE);
+
+    model.hardwareMaximumLineWidth = lineWidthRange[1];
```

Comment:
does not seem to make sense to "cache" it into model. Can you please use simple closure to cache it ?

Reply:
done!